HeterogeneousCore/SonicTriton/src/TritonClient.cc

```diff
@@ -363,6 +369,9 @@ void TritonClient::getResults(const std::vector<std::shared_ptr<tc::InferResult>
 void TritonClient::evaluate() {
   //undo previous signal from TritonException
   if (tries_ > 0) {
+    // Setup the service token for the current thread. So that we can access the

```

Comment:
I assume (because I don't remember well) the "current thread" here means a thread outside of framework's TBB worker thread pool. I think it would be good to clarify here (and probably also the place where `token_` is set) that the `evaluate()` is being called outside if framework's control, and therefore the service token is needed.

Reply:
For the record, the call chain is:
https://github.com/cms-sw/cmssw/blob/6c294992fd3129548f3608cce240808b8597faad/HeterogeneousCore/SonicCore/interface/SonicAcquirer.h#L39
https://github.com/cms-sw/cmssw/blob/6c294992fd3129548f3608cce240808b8597faad/HeterogeneousCore/SonicCore/interface/SonicClientBase.h#L31
https://github.com/cms-sw/cmssw/blob/6c294992fd3129548f3608cce240808b8597faad/HeterogeneousCore/SonicCore/src/SonicDispatcher.cc#L6
https://github.com/cms-sw/cmssw/blob/6c294992fd3129548f3608cce240808b8597faad/HeterogeneousCore/SonicTriton/src/TritonClient.cc#L363
https://github.com/cms-sw/cmssw/blob/6c294992fd3129548f3608cce240808b8597faad/HeterogeneousCore/SonicTriton/src/TritonClient.cc#L423
https://github.com/cms-sw/cmssw/blob/6c294992fd3129548f3608cce240808b8597faad/HeterogeneousCore/SonicTriton/src/TritonClient.cc#L454
https://github.com/cms-sw/cmssw/blob/6c294992fd3129548f3608cce240808b8597faad/HeterogeneousCore/SonicCore/src/SonicClientBase.cc#L45
https://github.com/cms-sw/cmssw/blob/6c294992fd3129548f3608cce240808b8597faad/HeterogeneousCore/SonicCore/src/SonicClientBase.cc#L51

`finish()` is called from inside Triton's `AsyncInferMulti()` function, and it can call `evaluate()` again if a retry is needed, so that second call can occur outside of the TBB thread pool.

Reply:
Just to be clear, I don't think these comments (in the code) need to explain the full call chain (thanks @kpedro88 for it anyway!). But I think explaining the "current" or "different" thread being a thread outside of the TBB thread pool and the `evaluate()` being called outside of the framework's control would be helpful for a future reader.

Reply:
Sure, I can update the comment

Reply:
I have updated to comments to clarify how the `evaluate` method can be called outside the TBB thread pool.